catroid/src/main/java/org/catrobat/catroid/content/actions/StoreCSVIntoUserListAction.kt

```diff
@@ -0,0 +1,145 @@
+
+        var columnToExtract = interpretColumnToExtract(formulaColumnToExtract)
+
+        columnToExtract--
+
+        val separator = extractOrReplaceSeperator(csvData)
+
+        val csvParser = CSVParserBuilder().withSeparator(separator).build()
+        val csvReader = CSVReaderBuilder(StringReader(csvData)).withCSVParser(csvParser)
+            .build()
+
+        val allRows = readCSVIntoList(csvReader)
+
+        if (allRows.size > 0 && columnToExtract >= 0) {
+            insertColumnValuesIntoUserList(allRows, columnToExtract)
+        }
+    }
+
+    fun interpretCSVData(formulaCSVData: Formula?): String {
+        return try {
+            formulaCSVData?.interpretString(sprite) ?: ""
+        } catch (exception: InterpretationException) {
+            Log.d(javaClass.simpleName, "Couldn't interpret formula", exception)
+            ""
+        }
+    }
+
+    fun interpretColumnToExtract(formulaColumnToExtract: Formula?): Int {
+        return try {
+            formulaColumnToExtract?.interpretInteger(sprite) ?: 0
+        } catch (exception: InterpretationException) {
+            Log.d(javaClass.simpleName, "Couldn't interpret formula", exception)
+            0
+        }
+    }
+
+    fun extractOrReplaceSeperator(csvData: String): Char {
+        var separator = ','
+        val pattern = Pattern.compile("^(?:\".*?\"|[^,;]*?)([,;])")
+        val matcher = pattern.matcher(csvData)
+        if (matcher.find()) {
+            separator = matcher.group(1)[0]
+        }
+        if (separator != ',' && separator != ';') {
+            separator = ','
+        }
+        return separator
+    }
+
+    fun readCSVIntoList(csvReader: CSVReader): List<Array<String>> {
+        return try {
+            csvReader.readAll()
+        } catch (exception: IOException) {
+            Log.e(javaClass.simpleName, "Couldn't read csv data", exception)
+            emptyList()
+        } catch (exception: CsvException) {
+            Log.e(javaClass.simpleName, "Couldn't validate csv data", exception)
+            emptyList()
+        }
+    }
+
+    fun insertColumnValuesIntoUserList(allRows: List<Array<String>>, columnToExtract: Int) {
+        var resetList = true
+        for (row in allRows) {
+            if (row.size <= columnToExtract) {
+                userList?.addListItem("")
+            } else {
+                resetList = false
+                userList?.addListItem(row[columnToExtract])
+            }
+        }
+        if (resetList) {
+            userList?.reset()
+        }
+    }
+
+    fun setUserList(userVariable: UserList) {
+        userList = userVariable
+    }
+
+    fun setFormulaColumnToExtract(formulaColumnToExtract: Formula) {
+        this.formulaColumnToExtract = formulaColumnToExtract
+    }
+
+    fun setFormulaCSVData(formulaCSVData: Formula) {
+        this.formulaCSVData = formulaCSVData
+    }
+
+    fun setSprite(sprite: Sprite) {
+        this.sprite = sprite
+    }
+}
```

Comment:
Well done!

PLEASE EXPLAIN: Does the changed behaviour affect the program? (After any Exception you return a default value and also the UserList reset is missing)
If this is not intended, I would suggest returning null and using "?: return" in the update method.

PLEASE ADD (if missing): a test for its original behaviour.